pkg/cache/cache.go

```diff
@@ -101,39 +104,59 @@ type FlavorInfo struct {
 	Labels     map[string]string
 }
 
-func NewClusterQueue(cap *kueue.ClusterQueue) *ClusterQueue {
+func NewClusterQueue(cap *kueue.ClusterQueue) (*ClusterQueue, error) {
 	c := &ClusterQueue{
-		Name:                 cap.Name,
-		RequestableResources: resourcesByName(cap.Spec.RequestableResources),
-		UsedResources:        make(Resources, len(cap.Spec.RequestableResources)),
-		Workloads:            map[string]*workload.Info{},
-		QueueingStrategy:     cap.Spec.QueueingStrategy,
+		Name:             cap.Name,
+		Workloads:        map[string]*workload.Info{},
+		QueueingStrategy: cap.Spec.QueueingStrategy,
 	}
+	if err := c.update(cap); err != nil {
+		return nil, err
+	}
+
+	return c, nil
+}
 
+func (c *ClusterQueue) update(in *kueue.ClusterQueue) error {
+	c.RequestableResources = resourcesByName(in.Spec.RequestableResources)
+	nsSelector, err := metav1.LabelSelectorAsSelector(in.Spec.NamespaceSelector)
+	if err != nil {
+		return err
+	}
+	c.NamespaceSelector = nsSelector
 	labelKeys := map[corev1.ResourceName]sets.String{}
-	for _, r := range cap.Spec.RequestableResources {
+	if c.UsedResources == nil {
+		c.UsedResources = make(Resources, len(in.Spec.RequestableResources))
+	}
+	for _, r := range in.Spec.RequestableResources {
 		if len(r.Flavors) == 0 {
 			continue
 		}
 
+		ts, exist := c.UsedResources[r.Name]
+		if !exist {
+			ts = make(map[string]int64, len(r.Flavors))
+		}
 		resKeys := sets.NewString()
-		ts := make(map[string]int64, len(r.Flavors))
 		for _, t := range r.Flavors {
+			if _, exist := ts[t.Name]; !exist {
+				ts[t.Name] = 0
```

Comment:
we need to cleanup the flavors that got removed

Reply:
done